File: 9-palindrome-number/solution.ts
function isPalindromeNumber_try1(x: number): boolean {
  // negative numbers can not be palindromes.
  if (x < 0) {
    return false;
  }

  const numStr = x.toString();

  let left = 0;
  let right = numStr.length - 1;

  while (left < right) {
    if (numStr[left] !== numStr[right]) {
      return false;
    }
    left++;
    right--;
  }

  return true;
}

console.log(isPalindromeNumber_try1(1211));

/**
 *
 */

const isPalindromeNumber_try2 = (x: number): boolean => {
  if (x < 0) {
    return false;
  }

  const original = x;
  let reversed = 0;

  while (x > 0) {
    const lastDigit = x % 10;
    reversed = reversed * 10 + lastDigit;
    x = Math.floor(x / 10);
  }

  return original === reversed;
};

console.log(isPalindromeNumber_try2(1211));

File: 9-palindrome-number/solution.go
package main

import (
	"fmt"
	"strconv"
)

// this try using strconv works.
func isPalindromeNumber(x int) bool {

	if x < 0 {
		return false
	}

	numStr := strconv.Itoa(x)
	left := 0
	right := len(numStr) - 1

	for left < right {
		if numStr[left] != numStr[right] {
			fmt.Println(false)
			return false
		}
		left++
		right--
	}
	fmt.Println(true)
	return true
}

// is palindrome try 2
func isPalindromeNumber_try2(x int) bool {
	if x < 0 {
		return false
	}

	original := x
	reversed := 0

	for x > 0 {
		lastDigit := x % 10
		reversed = reversed*10 + lastDigit

		x /= 10
	}

	fmt.Println(original == reversed)

	return original == reversed
}

func main() {
	fmt.Println("main running")

	isPalindromeNumber_try2(122)
}
